File: base/src/main/java/com/shrimp/base/widgets/dialog/ProgressDialog.java
package com.shrimp.base.widgets.dialog;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.TextView;

import com.shrimp.base.R;

import androidx.fragment.app.FragmentTransaction;

/**
 * 在每次进行dismiss的时候进行onDestroy()并设置为null
 * 因为在dismiss的时候，对应的DialogFragment已经移除了，会经过onDestroyView
 * 对应的Fragment经过销毁，所以对应持有的引用需要设置为null，否则会引起内存泄露
 */
public class ProgressDialog extends BaseDialog {

    private String mMessage;
    private boolean isShowing;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mContentView = inflater.inflate(R.layout.shrimp_view_loading, container, false);
        setDismissListener(() -> isShowing = false);
        return mContentView;
    }

    public boolean isShowing() {
        return isShowing;
    }

    @Override
    protected void getExtraParams() {

        Bundle bundle = getArguments();
        if (bundle == null)
            return;
        mMessage = bundle.getString("message");
    }

    public void setMessage(String text) {
        Bundle bundle = buildArguments();
        bundle.putString("message", text);
        setArguments(bundle);
    }

    @Override
    protected void initComponents(View view) {
        TextView tvMessage = view.findViewById(R.id.define_porgress_tvMessage);
        if (!TextUtils.isEmpty(mMessage)) {
            tvMessage.setText(mMessage);
        }
    }

    @Override
    public void onActivityCreated(Bundle arg0) {
        super.onActivityCreated(arg0);
        WindowManager.LayoutParams params = this.getDialog().getWindow().getAttributes();
        params.gravity = Gravity.CENTER;
//        params.gravity = Gravity.TOP;
//        params.y = GenericTools.dip2px(getActivity(), 50);
        params.dimAmount = 0.5f;
        this.getDialog().getWindow().setAttributes(params);
    }

    @Override
    public int show(FragmentTransaction transaction, String tag) {
        if (isShowing) return 0;
        try {
            isShowing = true;
            return super.show(transaction, tag);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return 0;
    }

    @Override
    public void dismiss() {
        if (!isShowing) return;
        try {
            super.dismiss();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: base/src/main/java/com/shrimp/base/widgets/dialog/ProgressDialog.kt
package com.shrimp.base.widgets.dialog

import android.os.Bundle
import android.text.TextUtils
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.FragmentTransaction
import com.shrimp.base.R

/**
 * Created by chasing on 2021/10/26.
 */
class ProgressDialog : BaseDialog() {

    private var mMessage: String? = null
    var isShowing = false
        private set

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mContentView = inflater.inflate(R.layout.shrimp_view_loading, container, false)
        setDismissListener(object : DismissListener {
            override fun dismiss() {
                isShowing = false
            }
        })
        return mContentView
    }

    override fun getExtraParams() {
        val bundle = arguments ?: return
        mMessage = bundle.getString("message")
    }


    fun setMessage(text: String?) {
        val bundle = buildArguments()
        bundle.putString("message", text)
        arguments = bundle
    }

    override fun initComponents(view: View?) {
        val tvMessage = view?.findViewById<TextView>(R.id.define_porgress_tvMessage)
        if (!TextUtils.isEmpty(mMessage)) {
            tvMessage?.text = mMessage
        }
    }

    override fun onActivityCreated(arg0: Bundle?) {
        super.onActivityCreated(arg0)
        val params = this.dialog!!.window!!.attributes
        params.gravity = Gravity.CENTER
        //        params.gravity = Gravity.TOP;
//        params.y = GenericTools.dip2px(getActivity(), 50);
        params.dimAmount = 0.5f
        this.dialog!!.window!!.attributes = params
    }

    override fun show(transaction: FragmentTransaction, tag: String?): Int {
        if (isShowing) return 0
        try {
            isShowing = true
            return super.show(transaction, tag)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return 0
    }

    override fun dismiss() {
        if (!isShowing) return
        try {
            super.dismiss()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}